astropy/wcs/tests/test_wcs.py

```diff
@@ -721,3 +726,17 @@ class NewWCS(wcs.WCS):
     with pytest.raises(TypeError) as exc:
         iter(w)
     assert exc.value.args[0] == "'NewWCS' object is not iterable"
+
+
+def test_sip_tpv_agreement():
+    sip_header = get_pkg_data_contents(
+        os.path.join("data", "siponly.hdr"), encoding='binary')
+    tpv_header = get_pkg_data_contents(
+        os.path.join("data", "tpvonly.hdr"), encoding='binary')
+
+    w_sip = wcs.WCS(sip_header)
+    w_tpv = wcs.WCS(tpv_header)
+
+    assert_array_almost_equal(
+        w_sip.all_pix2world([w_sip.wcs.crpix], 1),
+        w_tpv.all_pix2world([w_tpv.wcs.crpix], 1))
```

Comment:
This is comparing two files from the same data source -- one with the polynomial in TPV form and one in SIP form.  It is intended that they have the same transformation, even though the exact values in the coefficients are different due to the differences in conventions in TPV and SIP.  It's testing agreement between astropy's SIP and wcslib's TPV on different headers that are meant to match.  See @eteq's report here: https://github.com/astropy/astropy/pull/3845#issuecomment-111743742